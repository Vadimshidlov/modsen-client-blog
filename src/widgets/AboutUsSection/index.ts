export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
