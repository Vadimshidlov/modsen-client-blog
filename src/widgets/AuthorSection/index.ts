export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
