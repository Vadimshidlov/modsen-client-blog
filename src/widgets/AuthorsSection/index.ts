export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
