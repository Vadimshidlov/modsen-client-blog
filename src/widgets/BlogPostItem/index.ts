export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
