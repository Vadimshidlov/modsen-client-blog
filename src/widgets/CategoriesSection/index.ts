export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
