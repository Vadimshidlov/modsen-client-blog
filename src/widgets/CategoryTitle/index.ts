export { CategoryTitle } from '@/widgets/CategoryTitle/ui/CategoryTitle';
