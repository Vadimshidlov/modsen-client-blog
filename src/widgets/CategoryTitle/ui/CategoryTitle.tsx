import React from 'react';
import styles from '@/widgets/CategoryTitle/ui/CategoryTitle.module.scss';
import { Inter } from 'next/font/google';
import { getCapitalizeWord } from '@/shared/utils';
import { useTranslations } from 'next-intl';
import { CategoryTitlePropsType } from '@/shared/types';

const inter = Inter({
  subsets: ['latin'],
  weight: '600',
  variable: '--font-inter',
});

export function CategoryTitle({ categoryName }: CategoryTitlePropsType) {
  const t = useTranslations('Category');

  return (
    <section className={styles.categoryTitleContainer}>
      <h1 className={styles.categoryTitle}>{t(`${getCapitalizeWord(categoryName)}`)}</h1>
      <span className={styles.categoryDescription}>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
        labore.
      </span>
      <span className={`${styles.categoryPath} ${inter.variable}`}>
        BLOG &gt; {t(`${getCapitalizeWord(categoryName)}`).toUpperCase()}
      </span>
    </section>
  );
}
