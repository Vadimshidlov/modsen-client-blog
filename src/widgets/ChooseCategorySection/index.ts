export { ChooseCategorySection } from '@/widgets/ChooseCategorySection/ui/ChooseCategorySection';
