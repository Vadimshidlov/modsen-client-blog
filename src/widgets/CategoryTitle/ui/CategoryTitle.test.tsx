import React from 'react';
import { render } from '@testing-library/react';
import { CategoryTitle } from '@/widgets/CategoryTitle';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('SearchTags tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<CategoryTitle categoryName="Buisiness" />);
    expect(container).toMatchSnapshot();
  });

  test('renders category title correctly', () => {
    const { getByText } = render(<CategoryTitle categoryName="category" />);

    expect(getByText('Mocked Category')).toBeInTheDocument();
  });

  test('renders category description correctly', () => {
    const { getByText } = render(<CategoryTitle categoryName="category" />);

    expect(
      getByText(
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore.',
      ),
    ).toBeInTheDocument();
  });

  test('renders category path correctly', () => {
    const { getByText } = render(<CategoryTitle categoryName="category" />);

    expect(getByText('BLOG > MOCKED CATEGORY')).toBeInTheDocument();
  });
});
