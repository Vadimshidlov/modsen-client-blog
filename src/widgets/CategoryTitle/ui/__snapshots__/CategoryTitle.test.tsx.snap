// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTags tests should matches snapshot 1`] = `
<div>
  <section
    class="categoryTitleContainer"
  >
    <h1
      class="categoryTitle"
    >
      Mocked Buisiness
    </h1>
    <span
      class="categoryDescription"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore.
    </span>
    <span
      class="categoryPath variable"
    >
      BLOG &gt; 
      MOCKED BUISINESS
    </span>
  </section>
</div>
`;
